test/cpp/jit/test_gpu.cpp

```diff
@@ -4613,6 +4609,253 @@ void testGPU_FusionCacheMultiConsumer() {
       aten_output.sub(outputs[1]).abs().sum());
 }
 
+void testGPU_FusionSmem() {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  TensorView* tv0 = makeDummyTensor(2); // (M, N)
+  TensorView* tv1 = makeDummyTensor(2); // (M, N)
+  TensorView* tv2 = mul(tv0, tv1);
+  fusion.addInput(tv0);
+  fusion.addInput(tv1);
+  fusion.addOutput(tv2);
+  // Algorithm
```

Comment:
nit: same pattern, please use empty line, comment then the block to which the comment applies to
(general convention is to have comments precede the code blocks)